plugins/dell/fu-dell-common.c

```diff
@@ -326,13 +325,12 @@ fu_dell_toggle_flash (FuDevice *device, GError **error, gboolean enable)
 		if (!fu_device_has_flag (device, FWUPD_DEVICE_FLAG_UPDATABLE))
 			return TRUE;
 		tmp = fu_device_get_plugin (device);
-		if (!((g_strcmp0 (tmp, "thunderbolt") == 0) ||
```

Comment:
What happens if we're using new fwupd on an old kernel?

Reply:
I keep going back and forth on this.  The way the dell plugin works to force power is too unconditional and does't fit in the safely designed logic used here.  I'd like to have the dell plugin to fall back upon if possible.  If I do come up with a way to split this up to thunderbolt/thunderbolt-power with an intra-plugin API I'll retrofit the Dell plugin the same way instead.

Reply:
Since all of the (smarter) logic around when to force power and when not to lives in thunderbolt-power, what do you think about #ifdef(HAVE_DELL) in the new plugin to allow a fallback?  

Otherwise I think the the way to go is some sort of inter-plugin API with thunderbolt <-thunderbolt-power-> dell and duplicating most of the logic in thunderbolt-power in the dell plugin too.

Reply:
I'd much rather avoid `HAVE_DELL` if at all possible. What would you need to duplicate? I was under the impression that the dell plugin would be doing less, not more...

Reply:
Wild idea: could you use the new `CONFLICTS` thing and split out a dell_legacy plugin that does the tbt force power stuff on old kernels? Maybe we're just okay with requiring a new kernel for thunderbolt flashing things to work, but I've also got a neuron wired to the RHEL process too.

Reply:
Well the dell plugin really should have been following this approach (only power on when it's needed to rather than all the time).  I'm leaning on since you need a new kernel for thunderbolt flashing, one newer kernel version for thunderbolt at bootup working right, what's one newer one for thunderbolt force power?

Fedora & Ubuntu will quickly pick up a new kernel.
I think with RHEL the driver is contained enough that it should be backportable.  

I'm gonna go with kill the TBT force power from Dell plugin and say require this kernel interface instead.

Reply:
Works for me.